test/e2e/singlecluster/pod_test.go

```diff
@@ -0,0 +1,145 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/jobs/pod"
+	"sigs.k8s.io/kueue/pkg/util/kubeversion"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	podtesting "sigs.k8s.io/kueue/pkg/util/testingjobs/pod"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("Pod groups", func() {
+	var (
+		ns         *corev1.Namespace
+		onDemandRF *kueue.ResourceFlavor
+	)
+
+	ginkgo.BeforeEach(func() {
+		if kubeVersion().LessThan(kubeversion.KubeVersion1_27) {
+			ginkgo.Skip("Unsupported in versions older than 1.27")
+		}
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "pod-e2e-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		onDemandRF = testing.MakeResourceFlavor("on-demand").Label("instance-type", "on-demand").Obj()
+		gomega.Expect(k8sClient.Create(ctx, onDemandRF)).To(gomega.Succeed())
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandRF, true)
+	})
+
+	ginkgo.When("Single CQ", func() {
+		var (
+			cq *kueue.ClusterQueue
+			lq *kueue.LocalQueue
+		)
+
+		ginkgo.BeforeEach(func() {
+			cq = testing.MakeClusterQueue("cq").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "5").Obj(),
+				).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+			lq = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
+		})
+
+		ginkgo.It("should admit group that fits", func() {
+			g1 := podtesting.MakePod("g1", ns.Name).
+				Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"1s"}).
+				Queue(lq.Name).
+				Request(corev1.ResourceCPU, "1").
+				MakeGroup(2)
+			g1Key := client.ObjectKey{Namespace: ns.Name, Name: "g1"}
+			for _, p := range g1 {
+				gomega.Expect(k8sClient.Create(ctx, p)).To(gomega.Succeed())
+				gomega.Expect(p.Spec.SchedulingGates).
+					To(gomega.ContainElement(corev1.PodSchedulingGate{
+						Name: pod.SchedulingGateName}))
```

Comment:
This is just a question. Why the pod's schedulingGate is updated even though we didn't get the pod?
I guess that the pod is updated when creating the pod, right?

Reply:
Correct. Any API defaulting and mutations (via webhooks) comes back as part of the GET response.

Reply:
Thank you for the clarifications!